src/components/Form Fields/dropdown-headless.tsx

```diff
@@ -0,0 +1,121 @@
+import { Fragment } from "react";
+import { Menu, Transition } from "@headlessui/react";
+import {
+  FieldValues,
+  useController,
+  UseControllerProps
+} from "react-hook-form";
+import { HiCheck, HiChevronDown } from "react-icons/hi";
+
+import Label from "@/components/Form Fields/box-label";
+import Error from "@/components/Form Fields/error-msg";
+
+export interface FormProps<T extends FieldValues = FieldValues>
+  extends UseControllerProps<T> {
+  options: { id: number; text: string }[];
+  width?: string;
+  placeholder?: string;
+  label?: string;
+}
+
+function classNames(...classes: any) {
```

Comment:
Try defining the types of the params or just do the functionality of this inside the className prop